tests/test_target.py

```diff
@@ -14,35 +14,63 @@
 # limitations under the License.
 #
 
+"""Module for testing targets."""
+
 import os
 import shutil
 import tempfile
-from unittest import TestCase
+from pprint import pp
+import pytest
 
 from devlib import LocalLinuxTarget
+from devlib.utils.misc import load_struct_from_yaml
+
+
+def build_targets():
+    """Read targets from a YAML formatted config file"""
+
+    config_file = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'target_configs.yaml')
+
+    target_configs = load_struct_from_yaml(config_file)
+    if target_configs is None:
+        raise ValueError(f'{config_file} looks empty!')
+
+    targets = []
+
+    if target_configs.get('LocalLinuxTarget') is not None:
+        print('> LocalLinux targets:')
+        for entry in target_configs['LocalLinuxTarget'].values():
+            pp(entry)
+            ll_target = LocalLinuxTarget(connection_settings=entry['connection_settings'])
+            targets.append(ll_target)
+
+    return targets
+
 
+@pytest.mark.parametrize("target", build_targets())
+def test_read_multiline_values(target):
+    """
+    Test Target.read_tree_values_flat()
 
-class TestReadTreeValues(TestCase):
+    :param target: Type of target per :class:`Target` based classes.
+    :type target: Target
+    """
 
-    def test_read_multiline_values(self):
-        data = {
-            'test1': '1',
-            'test2': '2\n\n',
-            'test3': '3\n\n4\n\n',
-        }
+    data = {
+        'test1': '1',
+        'test2': '2\n\n',
+        'test3': '3\n\n4\n\n',
+    }
 
-        tempdir = tempfile.mkdtemp(prefix='devlib-test-')
-        for key, value in data.items():
-            path = os.path.join(tempdir, key)
-            with open(path, 'w') as wfh:
-                wfh.write(value)
+    tempdir = tempfile.mkdtemp(prefix='devlib-test-')
```

Comment:
This should use tempfile.TemporaryDirectory, as a context manager.

Reply:
Latest version of this code:
```Python
    with target.make_temp() as tempdir:
        ...
        raw_result = target.read_tree_values_flat(tempdir)
        result = {os.path.basename(k): v for k, v in raw_result.items()}
```
And this is `Target()::make_temp()` implementation (295f1269ed55c6a1c1b2a1c3b1e081f419931ed2):
```Python
    @contextmanager
    def make_temp(self, is_directory=True, directory='', prefix='devlib-test'):
        ...
        try:
            cmd = f'mktemp -p {directory} {prefix}-XXXXXX'
            if is_directory:
                cmd += ' -d'

            temp_obj = self.execute(cmd).strip()
            yield temp_obj
        finally:
            if temp_obj is not None:
                self.remove(temp_obj)
```